src/Table/read_fits.cxx

```diff
@@ -371,27 +371,25 @@ class Row_Column_Element_Reader {
 };
 //====================================================
 
-}  // namespace
+// Extract labeled_properties/trailing_info_lists/attributes.
+// If this table was created by write_fits(), those elements
+// would have been converted to labeled_properties and stored as keywords
+// of the form
 
+// ELEMENT.<prop_name>.XMLATTR.<attr_name> : attr_value or
+// ELEMENT.<prop_name>.XMLATTR.ATTR_IRSA_VALUE : prop.value
 
-void tablator::Table::read_fits(const boost::filesystem::path &path) {
-    CCfits::FITS fits(path.string(), CCfits::Read, false);
-    if (fits.extension().empty())
-        throw std::runtime_error("Could not find any extensions in this file: " +
-                                 path.string());
-    CCfits::ExtHDU &table_extension = *(fits.extension().begin()->second);
-    CCfits::BinTable *ccfits_table(dynamic_cast<CCfits::BinTable *>(&table_extension));
+// where <prop_name> is the value of prop's ATTR_NAME attribute and is
+// assumed to be non-empty for INFO elements, of which we might have many.
+//(We can't yet convert to FITS format VOTables with more than one RESOURCE.
+// 07Dec20)
 
-    static std::vector<std::string> fits_ignored_keywords{{"LONGSTRN"}};
-    static const auto keyword_ucd_mapping = fits_keyword_ucd_mapping(false);
+std::vector<std::pair<std::string, tablator::Property>>
+read_keywords_as_labeled_properties(CCfits::ExtHDU &table_extension) {
```

Comment:
So tabulator supports a fits file with a single HDU binary table.

Reply:
Good point.  As we start handling more FITS files, we might need to expand tablator's support for them.  Not in this PR, though.  Do you have other reservations about the current changes?